include/boost/url/detail/config.hpp

```diff
@@ -84,4 +86,9 @@
 # error Unknown or unsupported architecture, please open an issue
 #endif
 
+#ifndef BOOST_URL_DOCS
+// this is private
+using pos_t = size_t;
+#endif
```

Comment:
This could end up being a different size depending on platform, no? Why can't we do `unsigned int`? or uhh some guaranteed 32-bit integer type?